spec/unit/actions/app_apply_manifest_spec.rb

```diff
@@ -848,35 +849,182 @@ module VCAP::CloudController
               end
             end
 
-            context 'service binding errors' do
-              context 'bind happens async' do
-                context 'action starts async binding' do
-                  before do
-                    allow(service_cred_binding_create).to receive(:bind).and_return({ async: true })
+            context 'service broker support sync or async bindings' do
+              context 'action prefers sync binding' do
+                let(:binding1) { ServiceBinding.make(service_instance: service_instance, app: app) }
+                let(:binding2) { ServiceBinding.make(service_instance: service_instance_2, app: app) }
+
+                before do
+                  allow_any_instance_of(ServiceBinding).to receive(:terminal_state?).and_call_original
+                  allow_any_instance_of(AppApplyManifest).to receive(:sleep)
+
+                  precursor_count = 0
+                  allow(service_cred_binding_create).to receive(:precursor) do
+                    precursor_count += 1
+                    if precursor_count == 1
+                      binding1
+                    else
+                      binding2
+                    end
                   end
```

Comment:
Isn't this the same as `allow(service_cred_binding_create).to receive(:precursor).and_return(binding1, binding2)`?

See [Specify different return values for multiple calls](https://relishapp.com/rspec/rspec-mocks/docs/configuring-responses/returning-a-value#specify-different-return-values-for-multiple-calls)

Reply:
It's not quite the same--because `binding1` and `binding2` are in `let` calls, using `and_return(binding1, binding2)` will evaluate them immediately and insert records in the database, when we don't want the records to be created until we actually call this method.

Reply:
we may be able to use https://relishapp.com/rspec/rspec-mocks/v/3-12/docs/configuring-responses/mixed-responses if we prefer that syntax

Reply:
Ok, now I understand the difference. I did not know the "mixed responses" syntax, but I kind of like it. Maybe next time...